services/AttendanceService.ts

```diff
@@ -53,4 +55,27 @@ export default class AttendanceService {
       return attendance.getPublicAttendance();
     });
   }
+
+  public async addEventFeedback(feedback: string[], uuid: Uuid, user: UserModel): Promise<PublicAttendance> {
+    return this.transactions.readWrite(async (txn) => {
+      const attendanceRepository = Repositories.attendance(txn);
+      const event = await Repositories.event(txn).findByUuid(uuid);
+      if (!event) throw new NotFoundError('Event not found');
+      const hasAttended = await attendanceRepository.hasUserAttendedEvent(user, event);
+      if (!hasAttended) throw new UserError('You cannot provide feedback for an event you didn\'t attend');
+      const hasProvidedFeedback = await attendanceRepository.hasUserProvidedEventFeedback(user, event);
+      if (hasProvidedFeedback) throw new UserError('You cannot submit feedback for this event more than once');
```

Comment:
Don't need to make multiple queries for this. Grab the attendance object, check if it exists + check if `feedback` exists.